tests/kernel/gen_isr_table/src/main.c

```diff
@@ -100,14 +100,14 @@ ISR_DIRECT_DECLARE(isr2)
 }
 #endif
 
-void isr3(const void *param)
+void isr3(const struct device *param)
```

Comment:
For sure I'm missing something here. But why all the ISR parameters must necessarily be a `struct device`?